### DISCUSSION TITLE
VSCode icons

### DISCUSSION THREAD

**Discussion by prazdevs** [catppuccin/catppuccin#1942] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the requested port be themed?

👋  since i love the palette and am a user of the vscode theme (and iterm2 and such), I thought it'd be nice to have matching icons as well. I created a port based on icons I made (inspired by various icon themes).

Icons are made to be extremely consistent using only 5px wide strokes, and are automatically adjusted to fit the palette :)

### Has the requested port been themed already?

https://github.com/prazdevs/catppuccin-vscode-icons

### Any additional comments?

Feel free to hmu on discord @ PraZ

**Comment by backwardspy** [comment]
interested in @nekowinston's thoughts on including this with the vscode port versus keeping it separate.

edit: though i suspect it probably makes sense to have two separate extensions, regardless of the repo structure.

**Reply by nekowinston** [reply]
Totally agree with the edit. We can create an Extension Pack, like the [Python](https://marketplace.visualstudio.com/items?itemName=donjayamanne.python-extension-pack) pack.

**Reply by nekowinston** [reply]
As for the repo structure itself, I meant to include the converted Material Icons inside the catppuccin/vscode repo, since it already has automation for releasing VSCode Extensions set up. I'm generally for having them in a single repo, but if there are too many downsides to that, I'm open to having a separate repo as well.

**Comment by nekowinston** [comment]
Awesome work! I only had a quick look at most file in the repo, I haven't tried them out yet.

What's your opinion on adding color mappings to the icons as well? Currently you have icons mapped to file types, and then use DeltaE to generate the flavors - which is totally fine, that was my approach for my bodged Catpuccinified Material Icons as well.

The reason I held off on releasing those was because I wanted to first use Delta E as a "first pass" to generate most colors, then explicitly map Catppuccin colors to each file type. This would mean mapping the color `text` to `README.md`, instead of relying that the Latte DeltaE will make a guess that produces sufficient contrast.

**Reply by prazdevs** [reply]
thing is some icons have multiple colors 🤔 
i had a though on using css variables too, and changing these as a palette change.

**Comment by sgoudham** [comment]
This looks **amazing**, thanks for your hard work on this! It's now over for review at #1974
